contracts/contracts/RequestManager.sol

```diff
@@ -331,6 +331,22 @@ contract RequestManager is Ownable {
     {
         Claim storage claim = claims[claimId];
         Request storage request = requests[claim.requestId];
+
+        address claimReceiver = resolveClaim(claimId);
+
+        if (
+            request.depositReceiver == address(0) &&
+            claimReceiver == claim.claimer
+        ) {
+            withdrawDeposit(request, claim, claimReceiver);
+        }
+
+        return claimReceiver;
+    }
+
+    function resolveClaim(uint256 claimId) private returns (address) {
```

Comment:
I like the way you went with this. Can I suggest that we go one step further and make this function a `view` function?     
```
// Returns (deposit receiver, eth to transfer)
function resolveClaim(uint256 claimId) private view returns (address, uint256) {
```
That way, all the claim resolution logic would be contained in a single side-effect-free function, while the boring mechanics of sending ETH, emitting events etc. would be outside.